src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -54,52 +51,106 @@ public ContigAligner(final String referenceFilePath) throws IOException {
-
-
-                    final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
-                    if ( iterator.hasNext() ) {
-                        AlignmentRegion current = iterator.next();
-                        while ( iterator.hasNext() ) {
-                            final AlignmentRegion previous = current;
-                            current = iterator.next();
-                            String homology = "NA";
-                            if (previous.endInAssembledContig >= current.startInAssembledContig) {
-                                homology = new String(Arrays.copyOfRange(sequence, current.startInAssembledContig - 1, previous.endInAssembledContig));
-                            }
-
-                            String insertedSequence = "NA";
-                            if (previous.endInAssembledContig < current.startInAssembledContig - 1) {
-                                insertedSequence = new String(Arrays.copyOfRange(sequence, previous.endInAssembledContig, current.startInAssembledContig));
-                            }
-                            final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint(contigId, previous, current, homology, insertedSequence);
-                            assembledBreakpoints.add(assembledBreakpoint);
-                        }
-                    }
-                }
+                // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                        .filter(a -> a.getSecondary() < 0)
+                        .map(a -> new AlignmentRegion(breakpointId, contigId, a))
+                        .sorted(Comparator.comparing(a -> a.startInAssembledContig))
+                        .collect(arrayListCollector(alnRgns.length));
+                 alignedContigs.addAll(alignmentRegionList);
             }
         } catch (final IOException e) {
             throw new GATKException("could not execute BWA");
         }
 
-        return assembledBreakpoints;
+        return alignedContigs;
+    }
+
+    @VisibleForTesting
+    public static List<AssembledBreakpoint> getAssembledBreakpointsFromAlignmentRegions(final byte[] sequence, final List<AlignmentRegion> alignmentRegionList, final Integer minAlignLength) {
+        final List<AssembledBreakpoint> results = new ArrayList<>(alignmentRegionList.size() - 1);
+        final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
+        final List<String> insertionAlignmentRegions = new ArrayList<>();
+        if ( iterator.hasNext() ) {
+            AlignmentRegion current = iterator.next();
+            while (treatAlignmentRegionAsInsertion(current) && iterator.hasNext()) {
+                current = iterator.next();
+            }
+            while ( iterator.hasNext() ) {
```

Comment:
A very-unlikely edge case: right now if all `AlignmentRegion`'s in the input has mapping quality less than 60, then the while loop will fall through, and an empty list would be returned. Just in case.